### PR TITLE
Remove upper bound for python dependency

### PR DESCRIPTION
**Pull request by dudoslav**
Also increase lower bound to 3.9 since 3.8 is now "end of life"

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,12 +23,12 @@ build:
 
 requirements:
   host:
-    - python >=3.8,<3.12
+    - python >=3.9
```
Comment (johnkerl) [inline recipe/meta.yaml]: Proactively (just so we don't get bitten by it) let's day < 3.13

**Review by johnkerl — APPROVED** [review]

**Comment by jdblischak** [comment]
cellxgene-census 1.15, the current version of the recipe, requires python >= 3.8, < 3.12

https://github.com/chanzuckerberg/cellxgene-census/blob/v1.15.0/api/python/cellxgene_census/pyproject.toml#L14

We need to update to cellxgene-census 1.16.2, which requires python >= 3.10, < 3.13

https://github.com/chanzuckerberg/cellxgene-census/blob/v1.16.2/api/python/cellxgene_census/pyproject.toml#L14

**Comment by jdblischak** [comment]
When updating to cellxgene-census 1.16.2, we'll also need to update the tiledbsoma-py requirement (we can also remove the patch I added in #14)

https://github.com/chanzuckerberg/cellxgene-census/blob/v1.16.2/api/python/cellxgene_census/pyproject.toml#L29

```toml
dependencies= [
    # NOTE: the tiledbsoma version must be >= to the version used in the Census builder, to
    # ensure that the assets are readable (tiledbsoma supports backward compatible reading).
    # Make sure this version does not fall behind the builder's tiledbsoma version.
    "tiledbsoma>=1.12.3,!=1.14.1",
    "anndata",
    "numpy>=1.23,<2.0",
    "requests",
    "typing_extensions",
    "s3fs>=2021.06.1",
]
```

**Review by johnkerl — APPROVED** [review]

**Comment by jdblischak** [comment]
Superseded by #17

**Pull request closed without merging**